### DISCUSSION TITLE
Windows serial driver for ESP 8266

### DISCUSSION THREAD

**Discussion by Raja-Raman** [arendst/Tasmota#13656] [Q&A] [answered]
Sorry if this is off topic: But I am unable to upload Tasmota on my NodeMCU board ever since I upgraded my Windows 10. My old USB serial driver CH340 no longer works. Windows complains that the driver is 'not signed', so it cannot be installed. Windows Update also could not find any suitable driver for NodeMCU serial interface. The same board and cable work fine on a Linux laptop.
Everyone points me to a Chinese chip manufacturer's site to download the driver. But I am scared to download some unknown/ unsigned driver from PRC. A few drivers are available at the following places:
https://github.com/nodemcu/nodemcu-devkit/tree/master/Drivers
https://sparks.gogo.co.nz/ch340.html
But no one seems to know if these are safe or contain Trojans. If the driver is genuine, why doesn't the manufacturer sign it?
I just want a certified and safe serial driver for my ESP chips. Please help.

**Comment by Jason2866** [comment, marked as answer]
You find the official drivers on the manufacturer's site http://www.wch.cn/products/CH340.html?from=list

**Reply by sfromis** [reply]
You get no USB/serial driver for the ESP chips, you need drivers for each different USB/serial adapter chip (this is not part of ESP8266). Depends on which chip is on the board, seems that some manufacturers have started using CH9102F instead of the "usual suspects". Takes a different driver for the operating system in use.

If you have it working on Linux, the command dmesg (or whatever you want to display driver messages) can show you what was detected.
